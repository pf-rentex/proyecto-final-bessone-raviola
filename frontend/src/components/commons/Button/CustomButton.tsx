import React from 'react';
import PropTypes from 'prop-types';

interface IButtonProps {
  color?: "primary" | "secondary" | "alt";
  text: string;
  disabled?: boolean;
  callback?: Function;
  children?: React.ReactNode;
}

const CustomButton = ({
  color = "primary",
  text = "",
  disabled = false,
      callback = () => null,
      children = <></>,
    }: IButtonProps ) => {

  const onClicked = (e: React.MouseEvent<HTMLButtonElement>): void => {
    e.preventDefault();
    if (callback && !disabled) {
      callback();
    }
  }
  return (
      <button onClick={onClicked}
              className={`bg-${disabled ? 'gray-400' : color}${disabled ? 'cursor-not-allowed': ''} shadow-md w-full rounded-md px-5 py-2 flex content-center my-3 focus:shadow-sm outline-none align-items-center justify-center content-center align-middle justify-items-center items-center`}
    >
      <span className="text-2xl pr-4">{children}</span>
      <span
        className={`text-sm font-sm  text-${
          color === "primary" || color === "alt" ? "white" : "gray-800"
        } flex-1`}
      >
        {text}
      </span>
    </button>
  );
};

CustomButton.propTypes = {
  color: PropTypes.string,
  text: PropTypes.string.isRequired,
  children: PropTypes.node,
};

export default CustomButton;
