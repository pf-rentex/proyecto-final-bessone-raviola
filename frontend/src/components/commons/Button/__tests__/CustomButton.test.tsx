import React from 'react';
import {render} from '@testing-library/react';
import CustomButton from '../CustomButton';

describe('Custom button', () => {
  test('renders button', () => {
    const component = render(<CustomButton text='Hola' color="primary"/>);

    expect(component.container).toBeInTheDocument();
    expect(component.container).toMatchSnapshot();
  });

  test('renders button text', () => {
    const component = render(<CustomButton text='Hola' color="primary"/>);

    component.getByText('Hola');
  });
});
