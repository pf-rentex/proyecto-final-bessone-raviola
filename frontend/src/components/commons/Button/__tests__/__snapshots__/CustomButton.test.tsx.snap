// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom button renders button 1`] = `
<div>
  <button
    class="bg-primary  shadow-lg w-full rounded-md px-5 py-2 flex content-center my-1 focus:shadow-sm font-secondary font-semibold outline-none align-items-center justify-center content-center align-middle justify-items-center items-center "
  >
    <span
      class="text-2xl "
    />
    <span
      class="text-sm font-sm  text-white flex-1"
    >
      Hola
    </span>
  </button>
</div>
`;
